testsupport/space.go

```diff
@@ -87,13 +87,26 @@ func WithTierNameAndHashLabel(tierName, hash string) SpaceOption {
 // CreateSpace initializes a new Space object using the NewSpace function, and then creates it in the cluster
 // It also automatically provisions MasterUserRecord and creates SpaceBinding for it
 func CreateSpace(t *testing.T, awaitilities wait.Awaitilities, opts ...SpaceOption) (*toolchainv1alpha1.Space, *toolchainv1alpha1.UserSignup, *toolchainv1alpha1.SpaceBinding) {
+	// we need to create a MUR & SpaceBinding, otherwise, the Space could be automatically deleted by the SpaceCleanup controller
+	username := uuid.Must(uuid.NewV4()).String()
+	signup, mur := NewSignupRequest(awaitilities).
+		Username(username).
+		Email(username + "@acme.com").
+		ManuallyApprove().
+		RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+		NoSpace().
+		WaitForMUR().Execute(t).Resources()
+	t.Logf("The UserSignup %s and MUR %s were created", signup.Name, mur.Name)
+	// then create the actual space
 	space := NewSpace(t, awaitilities, opts...)
 	err := awaitilities.Host().CreateWithCleanup(t, space)
 	require.NoError(t, err)
+	// create spacebinding request immediately after
+	spaceBinding := CreateSpaceBinding(t, awaitilities.Host(), mur, space, "admin")
```

Comment:
An alternative to this logic, could be to try and issue a dryRun creation of the space, so that kube api server returns an actual unique space object generated name, and use that name for the usersignup and spacebinding, and only after those two are created trigger the actual space creation request.

Reply:
Let's stick with the current solution. If it doesn't help then we could try to use dryRun...